server/utils/socketsServer/socketServer.js

```diff
@@ -0,0 +1,91 @@
+const socketio = require("socket.io");
+
+const connectedUsers = {};
+let socket = null;
+let io = null;
+
+const initSocketServer = (server, req) => {
+
+  if(!io) {
+    io = socketio(server, {
+      cors: {
+        origin: "*"
+      }
+    });
+    req.io = io;
+  }
+
+  io.on("connection", _socket => {
```

Comment:
This event is probably where it makes the most sense to have the userJoin event, just make sure that the socket is reconnecting on login/sign up from the frontend